src/components/app/data/utils.js

```diff
@@ -747,3 +747,54 @@ export function isEnrollmentUpgradeable(enrollment) {
   const canUpgradeToVerifiedEnrollment = enrollment.mode === COURSE_MODES_MAP.AUDIT && !isEnrollByLapsed;
   return canUpgradeToVerifiedEnrollment;
 }
+
+export function determineAllocatedCourseRuns({
+  redeemableLearnerCreditPolicies,
+  courseKey,
+}) {
+  const { learnerContentAssignments } = redeemableLearnerCreditPolicies;
+  if (learnerContentAssignments.hasAllocatedAssignments) {
+    let allocatedCourseRunAssignments = learnerContentAssignments.allocatedAssignments.filter(
+      (assignment) => assignment?.isAssignedCourseRun,
+    );
+    if (courseKey) {
```

Comment:
[curious/clarification] When is this function called _without_ a `courseKey` present? Reviewing the code paths that call this function seems like the `courseKey` is expected to be a valid string here, in both the `courseLoader` and `useCourseMetadata`.

For example, in the open PR migrating `courseLoader ` to TypeScript, it explicitly types `courseKey` as always being a string for this loader. Regarding `useCourseMetadata`, the related query is not `enabled` unless `courseKey` is present.

Reply:
at the moment not at all but see this related comment
> Sooo this code snippet was an earlier implementation, but opted to do a conditional on courseKey. The conditional on courseKey allows us to generalize this function to act as a helper for all allocated course run based assignments regardless of whether or not an actual courseKey is passed. This could be useful down the road but happy to go back to the original implementation as well 😄